packages/astro/src/@types/astro.ts

```diff
@@ -1142,10 +1147,12 @@ export interface AstroIntegration {
 		'astro:config:setup'?: (options: {
 			config: AstroConfig;
 			command: 'dev' | 'build';
+			isConfigReload: boolean;
 			updateConfig: (newConfig: Record<string, any>) => void;
 			addRenderer: (renderer: AstroRenderer) => void;
 			injectScript: (stage: InjectedScriptStage, content: string) => void;
 			injectRoute: (injectRoute: InjectedRoute) => void;
+			injectWatchTarget: (target: InjectedWatchTarget) => void;
```

Comment:
nit: the inject term seems in line with the other inject API, but I wonder if we could name it as `addWatchFile` so it's similar to Rollup's API, and maybe only accept an absolute string path to simplify the api 🤔

Reply:
I agree using `addWatchFile`. We can declare `addWatchFile: (path: string) => void;` and, if we see that using relative paths is common, we could add `addWatchFile: (path: string, relative?: boolean) => void;` or something like that